docs/minimal-metta.md

```diff
@@ -171,6 +171,35 @@ separate alternative.
 can collect the list of alternatives using `collapse-bind` filter them and
 return filtered items to the plan using `superpose-bind`.
 
+## Scope of a variable
+
+Variable name should be unique inside a separate expression evaluated. The
+reason is that the whole expression is a variable scope. For example one can
+write the expression `(chain (unify $parent Bob () ()) $_ $parent)`. And value
+of the `$parent` is returned correctly.
+
+When variable is passed as an argument to the function call and matched by a
+value then the value is assigned to the variable. If variable passed as an
+actual argument is matched by a formal argument variable then it is referenced
```

Comment:
I feel like the terms "formal argument" and "actual argument" need to be defined.  The example helps but I found this part a little bit confusing.

Reply:
It is a quite common terminology (for example see [Wikipedia article](https://en.wikipedia.org/wiki/Parameter_(computer_programming))). Is using word "argument" for the passed arguments and word "parameter" for the function input variables clears the thing? For example:
```
When a variable is passed as an argument to a function call and matched by a
value used in a function definition then the value is assigned to the variable.
If variable passed as an argument is matched by a function's parameter variable
then parameter variable references the variable passed as an argument. In this
case the variable passed as an argument can receive a value outsides of its
scope.
```

Reply:
Ah ok.  Thanks.  I had never encountered that phrasing before.

Reply:
Well it is not only terminology and I started using it intentionally to make readers clearly see that both sides are _arguments_ but _different arguments_. I am not sure if it is easy to read this. If it is hard and there is more familiar words then I would use them.